androidsdk/src/main/java/com/wootric/androidsdk/SurveyManager.java

```diff
@@ -356,24 +358,20 @@ void setAccessToken(String accessToken) {
         this.accessToken = accessToken;
     }
 
-    private String getOriginUrl() {
+    private String getOriginUrl(Activity activity) {
         if(originUrl == null) {
             PackageManager pm;
             String packageName;
             ApplicationInfo appInfo;
 
             try {
-                if (fragmentActivity != null) {
-                    pm = fragmentActivity.getPackageManager();
-                    packageName = fragmentActivity.getApplicationInfo().packageName;
-                } else {
-                    pm = activity.getPackageManager();
-                    packageName = activity.getApplicationInfo().packageName;
-                }
+                pm = activity.getPackageManager();
+                packageName = activity.getApplicationInfo().packageName;
                 appInfo = pm.getApplicationInfo(packageName, 0);
                 originUrl = pm.getApplicationLabel(appInfo).toString();
             } catch (Exception e) {
                 Log.e(Constants.TAG, "getOriginUrl: " + e.toString());
+                resetSurvey();
```

Comment:
I would remove this call to `resetSurvey();`. In the function `showSurveyFragment()`, the result of the call to `getOriginUrl()` doesn't matter: whatever value it returns, the fragment will still be created and launched, so the `surveyRunning` and `currentEvent` attributes should remain unchanged. It might encounter a problem later on, but that's up to the fragment to handle it gracefully.